components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretsProcessor.java

```diff
@@ -31,22 +31,22 @@
      * @param object Name of the {@link T}.
      * @throws SecretManagementException Secret management exception.
      */
-    T associateSecrets(T object) throws SecretManagementException;
+    T decryptAssociatedSecrets(T object) throws SecretManagementException;
 
     /**
      * This API is used to store encrypted secrets to the IDN_SECRET table.
      *
      * @param object Name of the {@link T}.
      * @throws SecretManagementException Secret management exception.
      */
-    T addOrUpdateSecrets(T object) throws SecretManagementException;
+    T encryptAssociatedSecrets(T object) throws SecretManagementException;
```

Comment:
Update the method descriptions as well.

Reply:
Resolved in [7f035fe](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/7f035fedca5ed28a27f590804ea2f47e1042da4d)